cockatrice/src/carddatabase.cpp

```diff
@@ -194,23 +196,35 @@ void SetList::markAllAsKnown()
 
 void SetList::guessSortKeys()
 {
-    // sort by release date DESC; invalid dates to the bottom.
-    QDate distantFuture(2050, 1, 1);
-    int aHundredYears = 36500;
+    defaultSort();
     for (int i = 0; i < size(); ++i) {
         CardSetPtr set = at(i);
         if (set.isNull()) {
             qDebug() << "guessSortKeys set is null";
             continue;
         }
+        set->setSortKey(i);
+    }
+}
 
-        QDate date = set->getReleaseDate();
-        if (date.isNull()) {
-            set->setSortKey(static_cast<unsigned int>(aHundredYears));
-        } else {
-            set->setSortKey(static_cast<unsigned int>(date.daysTo(distantFuture)));
+void SetList::defaultSort()
+{
+    std::sort(begin(), end(), [](const CardSetPtr &a, const CardSetPtr &b) {
+        QDate distantPast = QDate(1970, 1, 1);
+        QDate aDate = a->getReleaseDate().isValid() ? a->getReleaseDate() : distantPast;
+        QDate bDate = b->getReleaseDate().isValid() ? b->getReleaseDate() : distantPast;
+        CardSet::Priority aPriority = a->getPriority() ? a->getPriority() : CardSet::PriorityFallback;
+        CardSet::Priority bPriority = b->getPriority() ? b->getPriority() : CardSet::PriorityFallback;
```

Comment:
setting the fallback to be pretty low causes most custom sets and spoilers to be sorted pretty late, it might be nice to have custom sets be sorted early, it's a more advanced feature though but most users of custom sets would like them prioritised. the spoilers being sorted can be added to https://github.com/Cockatrice/Magic-Spoiler

Reply:
Thanks for the repo link for spoilers; I'll look into making a PR.

Regarding custom sets: having experimented with them a bit now, it looks like Cockatrice doesn't persist whether a set is custom or not, making it impossible to factor into sorting. That said, the default sort function _does_ respect the priority defined in the custom set's XML, including priorities with greater precedence than Core/Expansion. So, my proposal is to simply not distinguish custom sets, and document the priority field for custom set creators to incorporate it into their XML files. As you say, it's an advanced feature, so it doesn't seem like an unreasonable ask.

The other option I see would be to persist a flag that distinguishes custom sets from official sets, and then use that flag to push custom sets to the top, but I didn't want to go down that road before discussing it. Let me know what you think.

Reply:
with this pr all sets that don't have a priority set will be either custom sets or the spoiler xml, we can change the spoiler xml generation to fix the last one, so we do have a way to tell if a set is custom, it'd simply be all sets without priority set

Reply:
in case new set types get added, I propose we do change the oracle code for those to simply set them to a low type

Reply:
I made `PriorityFallback` 1, above the others.

Reply:
To me it seems worthwhile to distinguish between a set with a priority that hasn't been evaluated yet (`PriorityUndefined`; zero; falsy), and a set that has been evaluated and found no matching priority (`PriorityFallback`; 1; truthy, like other prioritized sets).

Reply:
That said, it's not something I feel strongly about and am happy to change it to zero. It just made the most sense in the moment to preserve the distinction, since the comments were only regarding where `PriorityFallback` falls, not the possibility of collapsing `PriorityFallback` and `PriorityUndefined` together.

Reply:
what benefits would having fallback over undefined have? it always gets changed from one into the other and there don't seem to be any other checks for it.

Reply:
Removed `PriorityUndefined` and set `PriorityFallback` to 0.